src/common/mfu_flist_copy.c

```diff
@@ -1554,50 +1587,91 @@ static int mfu_is_eof(const char* file, int fd)
+    mfu_file_t* mfu_dst_file)
 {
-    /* hint that we'll read from file sequentially */
-//    posix_fadvise(in_fd, offset, chunk_size, POSIX_FADV_SEQUENTIAL);
-
+    /* There is no equivalent "lseek" in daos, it is part of read, so skip that
+     * for a DAOS path */
     /* seek to offset in source file */
-    if(mfu_lseek(src, in_fd, offset, SEEK_SET) == (off_t)-1) {
-        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in source path `%s' (errno=%d %s)",
-            src, errno, strerror(errno));
+    if(mfu_file_lseek(src, mfu_src_file, offset, SEEK_SET) == (off_t)-1) {
+        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in source path `%s' (errno=%d %s)", src, errno, strerror(errno));
         return -1;
     }
 
     /* seek to offset in destination file */
-    if(mfu_lseek(dest, out_fd, offset, SEEK_SET) == (off_t)-1) {
-        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in destination path `%s' (errno=%d %s)",
-            dest, errno, strerror(errno));
+    if (mfu_file_lseek(dest, mfu_dst_file, offset, SEEK_SET) == (off_t)-1) {
+        MFU_LOG(MFU_LOG_ERR, "Couldn't seek in destination path `%s' (errno=%d %s)", dest, errno, strerror(errno));
         return -1;
     }
 
-    /* get buffer */
+    /* declare daos vars */
+#ifdef DAOS_SUPPORT
+    daos_size_t got_size;
+    d_sg_list_t sgl;
+    d_iov_t     iov;
+#endif
+
+    /* set buffer and buffer size */
     size_t buf_size = mfu_copy_opts->block_size;
     void* buf = mfu_copy_opts->block_buf1;
+    void* newbuf;
+
+    /* set daos vars and buffers */
+#ifdef DAOS_SUPPORT
+    if (mfu_src_file->type == DAOS || mfu_dst_file->type == DAOS) {
+        sgl.sg_nr = 1;
+        D_ALLOC(buf, buf_size);
+        D_ASSERT(buf != NULL);
+        d_iov_set(&iov, buf, buf_size);
+        sgl.sg_iovs = &iov;
+        sgl.sg_nr_out = 1;
+        if (mfu_src_file->type == DAOS) {
+            mfu_src_file->offset = (daos_off_t) offset;
+            mfu_src_file->sgl = &sgl;
+        }
+        if (mfu_dst_file->type == DAOS) {
+            mfu_dst_file->offset = (daos_off_t) offset;
+            mfu_dst_file->sgl = &sgl;
+        }
+    }
+#endif
 
     /* write data */
     size_t total_bytes = 0;
-    while(total_bytes < (size_t)length) {
+    char* buffer = mfu_copy_opts->block_buf1;
+    while(total_bytes <= (size_t)length) {
         /* determine number of bytes that we
          * can read = max(buf size, remaining chunk) */
         size_t left_to_read = (size_t)length - total_bytes;
+        
         if(left_to_read > buf_size) {
+#ifdef DAOS_SUPPORT
+            if (mfu_src_file->type == DAOS || mfu_dst_file->type == DAOS) {
+                newbuf = mfu_copy_opts->block_buf1;
+  	        D_REALLOC(newbuf, buf, left_to_read);
+                d_iov_set(&iov, newbuf, left_to_read);
+            }
+#endif
             left_to_read = buf_size;
         }
+
         if(mfu_copy_opts->synchronous) {
             /* O_DIRECT requires particular read sizes */
             left_to_read = buf_size;
         }
```

Comment:
I haven't thought about this change that I added with respect to what DAOS needed.  Is this new logic flow that computes the size of the read still valid for DAOS?